test/test.bats

```diff
@@ -83,7 +83,7 @@ execdirect() {
 @test "shebang-direct" {
   execdirect ./shebang.sh foo bar1
   assert_success
-  assert_output "./printargs ./shebang.sh foo bar1"
+  assert_output "./printargs /work/shebang.sh foo bar1"
```

Comment:
@tonistiigi About this change: https://github.com/tonistiigi/binfmt/actions/runs/5305194882/jobs/9602044053#step:4:385

```
#43 15.22 not ok 11 shebang-direct
#43 15.22 # (from function `assert_output' in file bats-assert/src/assert.bash, line 239,
#43 15.22 #  in test file test.bats, line 86)
#43 15.23 #   `assert_output "./printargs ./shebang.sh foo bar1"' failed
#43 15.23 #
#43 15.23 # -- output differs --
#43 15.23 # expected : ./printargs ./shebang.sh foo bar1
#43 15.23 # actual   : ./printargs /work/shebang.sh foo bar1
#43 15.23 # --
#43 15.23 #
```

linux-user now resolves the full path name of the started process at startup. See https://github.com/qemu/qemu/commit/258bec39f3e58fcdb006028e5c5a1801136ef04a

Reply:
Isn't this a breaking change (at least for the direct mode without `-P`)? What if the program reads its `args[0]`, it wouldn't expect that the value gets modified.

If this is how it behaves with `QEMU_PRESERVE_ARGV0` in kernel then it just looks plain wrong. If it is just a problem with our direct exec patches then the patches may need updates.

Reply:
@hdeller are we missing some context in here?